chainercv/datasets/cityscapes/cityscapes_semantic_segmentation_dataset.py

```diff
@@ -0,0 +1,108 @@
+import glob
+import os
+
+import numpy as np
+
+from chainer import dataset
+from chainer.dataset import download
+from chainercv.datasets.cityscapes.cityscapes_utils import cityscapes_labels
+from chainercv.utils import read_image
+
+
+class CityscapesSemanticSegmentationDataset(dataset.DatasetMixin):
+
+    """Dataset class for a semantic segmentation task on `Cityscapes dataset`_.
+
+    .. _`Cityscapes dataset`: https://www.cityscapes-dataset.com
+
+    .. note::
+
+        Please manually downalod the data because it is not allowed to
+        re-distribute Cityscapes dataset.
+
+    Args:
+        data_dir (string): Path to the dataset directory. The directory should
+            contain at least two directories, :obj:`leftImg8bit` and either
+            :obj:`gtFine` or :obj:`gtCoarse`. If :obj:`None` is given, it uses
+            :obj:`$CHAINER_DATSET_ROOT/pfnet/chainercv/cityscapes` by default.
+        label_mode ({'fine', 'coarse'}): The resolution of the labels. It
```

Comment:
If it's `label_dtype`, it surely sounds like a datatype, but I think `label_type` is OK. Anyway, just keeping the consistency of naming from the official scripts is better I think.